src/inmanta/agent/executor.py

```diff
@@ -45,7 +45,14 @@
 
 LOGGER = logging.getLogger(__name__)
 
-FailedResourcesSet: typing.TypeAlias = set[ResourceType]
+
+@dataclass(frozen=True)
+class FailedResource:
+    resource_type: ResourceType
+    exception: Exception
+
+
+FailedResourcesSet: typing.TypeAlias = set[FailedResource]
```

Comment:
I wonder if this shouldn't be a dict. As it is now, keeping duplicates (same type, different exception) out is not trivial.